### ISSUE TITLE
Consider renaming resolve_components to resolved_components

### ISSUE THREAD

**Issue by hsivonen** [unicode-org/icu4x#4042]
The documentation says it returns what was resolved, but the name suggests triggering resolution.

**Comment by Manishearth** [comment]
This is referring to the [experimental DateTimeFormatter API](https://unicode-org.github.io/icu4x/docs/icu_datetime/struct.DateTimeFormatter.html#method.resolve_components).

A bunch of this will probably change under skeletons anyway.

@zbraniecki to approve.

**Comment by hsivonen** [comment]
Also, per discussion elsewhere, the return type should probably be some kind of `ResolvedBag` that doesn't wrap each field in `Option`.

**Comment by zbraniecki** [comment]
Agree. And agree with the last comment, the `ResolvedBag` will come from `icu_preferences`.

**Comment by sffc** [comment]
This API will be changing in 2.0 but I'll make sure to call it `resolved_options`.

**Comment by sffc** [comment]
Datetime has a new API for this, which is documented. Might still need to look at other components

**Comment by Manishearth** [comment]
Collator is renamed to resolved_options.

DateTime uses https://unicode-org.github.io/icu4x/rustdoc/icu/datetime/pattern/struct.DateTimePattern.html, a different API.
